### ISSUE TITLE
The inspection reports of case hardcoded booleans

### ISSUE THREAD

**Issue by almost-online** [kalessil/phpinspectionsea#1305]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections ​(EA Extended)​, v3.0.13 |
| Language level | e.g. PHP 7.1                       |

Current behaviour
--
![image](https://user-images.githubusercontent.com/5806935/56595600-96a16d80-65f7-11e9-8733-d4508c269385.png)
```php
    public function saveData(array $data): bool
    {
        $connection = $this->getConnection();
        $connection->beginTransaction();

        // should return true when save data and commit or false
        return ($this->saveObject($data) && $connection->commit()) or ($connection->rollBack() && false);
    }
```

Expected behaviour
--

Environment details
---

PhpStorm 2019.1.1
Php Inspections ​(EA Extended)​, v3.0.13
Build #PS-191.6707.66, built on April 17, 2019
JRE: 1.8.0_202-release-1483-b44 amd64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
Windows 10 10.0

**Comment by funivan** [comment]
Hello, @almost-online Is this a problem? 
You should not write true or false in the source code because this makes no sense.

**Comment by kalessil** [comment]
@almost-online: can you share more similar samples please? 
PS: Were other reported cases correctly spotted?

I somehow agree with @funivan , but would like to better understand your case as well.

**Comment by almost-online** [comment]
this is not problem but I should to rewrite the code to this (equal) one:
```php
    public function saveData(array $data): bool
    {
        $connection = $this->getConnection();
        $connection->beginTransaction();

        // should return true when save data and commit or false
       if ($this->saveObject($data)) {
           return $connection->commit();
       }

        $connection->rollBack(); 
        return false;
    }
```

**Comment by kalessil** [comment]
Ok, then one more question: was the refactoring a good thing in your case?

**Comment by almost-online** [comment]
in case `($connection->rollBack() && false)` - the `$connection->rollBack()` will return `true` when rolled back the transaction but I shoudl always return `false` in case I`ve not save an object

**Comment by almost-online** [comment]
> Ok, then one more question: was the refactoring a good thing in your case?
probably this refactor will become mo readable but the code become longer

**Comment by almost-online** [comment]
anyway, this is not urgent

**Comment by funivan** [comment]
Refactored code seems to be better for me ;) Easy to read and maintain. IMHO

**Comment by kalessil** [comment]
Implemented the new settings!
